docs/gates.md

```diff
@@ -67,11 +67,16 @@ print(cirq.unitary(sqrt_x))
 The Pauli gates included in Cirq use the convention ``Z**0.5 ≡ S ≡ np.diag(1, i)``, ``Z**-0.5 ≡ S**-1``, ``X**0.5 ≡ H·S·H``, and the square root of ``Y`` is inferred via the right hand rule.
 Note that it is often the case that ``(g**a)**b != g**(a * b)``, due to the intermediate values normalizing rotation angles into a canonical range.
 
-#### KnownMatrix
+#### `cirq.unitary` and `def _unitary_` 
 
 We've seen this above.
-These are ``Gate`` or ``Operation`` instances which implement the ``matrix`` method.
-This returns a numpy ``ndarray`` matrix which is the unitary gate for the gate/operation.
+These are ``Gate`` or ``Operation`` instances which may be described by a
+unitary matrix.
+They implement the ``_unitary_`` method,
+which returns a numpy ``ndarray`` matrix which is the unitary gate for the
+gate/operation.
+The method may also return `NotImplemented`, in which case `cirq.unitary`
+behaves as if the method is not implemented.
```

Comment:
Might want to explain what actually happens, e.g. "in which case `cirq.unitary` raises a type error or returns a default value if provided."